### DISCUSSION TITLE
Support RFC 8229: TCP Encapsulation of IKE and IPsec Packets

### DISCUSSION THREAD

**Discussion by csszep** [strongswan/strongswan#1566] [Ideas]
Hi!

There is an open  issue in the old? wiki:

https://wiki.strongswan.org/issues/2189

Is there a plan to implement it?

Linux kernel supports ESP in TCP since 5.6

https://lwn.net/Articles/811198/

**Comment by tobiasbrunner** [comment]
There are currently no plans to do so. If anybody is interested in sponsoring this feature, please contact us.

**Reply by TomHsiung** [reply]
A-ha! The Gemini told me this is already available.

**Comment by TomHsiung** [comment]
Here I searched and found an opensourece package called "Libreswan" which supports the RFC 8229. Maybe their code is helpful for strongSwan to add this TCP-based encapsulation in future.

Link: [https://libreswan.org/wiki/RFC_8229_-_TCP_support_for_IKEv2_and_ESP](url)
